lesson22/index.js

```diff
@@ -0,0 +1,197 @@
+    const th = createElementWithClassName("th", "js-th");
+    th.textContent = items[key];
+    th.id = `js-${key}`;
+    tr.appendChild(th);
+  });
+  thead.appendChild(tr);
+  return thead;
+};
+
+const createTableBody = (usersData, keys) => {
+  const tbody = document.createElement("tbody");
+  usersData.forEach((userData) => {
+    const tr = createElementWithClassName("tr", "js-tr-inTbody");
+    tbody.appendChild(tr).appendChild(createTd(userData, keys));
+  });
+  return tbody;
+};
+
+const createTd = (usersData, keys) => {
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i < keys.length; i++) {
+    const td = document.createElement("td");
+    td.textContent = usersData[keys[i]];
+    fragment.appendChild(td);
+  }
+  return fragment;
+};
+
+const addSortButton = () => {
+  const sortItemsId = ["userId", "age"];
+  sortItemsId.forEach((id) => {
+    document.getElementById(`js-${id}`).appendChild(createSortButton());
+  });
+}
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName(
+    "button",
+    "sort-button js-sort-button"
+  );
+  sortButton.dataset.sortStatus = "default";
+  return sortButton;
+};
+
+const setClickInSortButton = () => {
+  const defaultRows = [...document.querySelectorAll(".js-tr-inTbody")];
+  const sortButtons = [...document.querySelectorAll(".js-sort-button")];
+
+  sortButtons.forEach((sortButton) => {
+    sortButton.addEventListener("click", (e) => {
+      resetSortButtonsExceptTarget(sortButtons, e.target);
+      const nextStatus = switchSortStatus(e.target.dataset.sortStatus);
+      e.target.dataset.sortStatus = nextStatus;
+      const sortedRows = getSortedRows(nextStatus, defaultRows, e.target);
+      const tbody = document.querySelector("tbody");
+      sortedRows.forEach((row) => {
+        tbody.appendChild(row);
+      });
+    });
+  });
+};
+
+const resetSortButtonsExceptTarget = (sortButtons, target) => {
+  sortButtons.filter((button) => button !== target).forEach((value) => {
+    value.dataset.sortStatus = "default";
+  })
+}
+
+const switchSortStatus = (status) => {
+  switch (status) {
+    case "default":
+      return "asc";
+
+    case "desc":
+      return "default";
+
+    case "asc":
+      return "desc";
+
+    default:
+      return "default";
+  }
+};
+
+const findClickedCellIndex = (target) =>
+  [...document.querySelectorAll(".js-th")].indexOf(target.parentElement);
+
+const getSortedRows = (status, defaultRows, target) => {
+  if (status === "default") return defaultRows;
+  const index = findClickedCellIndex(target);
+  switch (status) {
+    case "asc":
+      return [...defaultRows].sort(
+        (a, b) => a.children[index].textContent - b.children[index].textContent
+      );
+    case "desc":
+      return [...defaultRows].sort(
+        (a, b) => b.children[index].textContent - a.children[index].textContent
+      );
+    default:
+      throw new Error(`${status} is not provided.`);
+  }
+};
+
```

Comment:
I've never created a sort function before, but I didn't know there was such a way! I thought your code was very clear.

Reply:
Thanks for the review.

One of my goals was to write code that was easy for others to understand when they saw it. Your words give me confidence.☺️